api/pkg/artifact/artifact.go

```diff
@@ -2,31 +2,121 @@ package artifact
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/url"
 	"strings"
 
 	"cloud.google.com/go/storage"
 	"google.golang.org/api/iterator"
 )
 
+// URL contains the information needed to identify the location of an object
+// located in Google Cloud Storage.
+type URL struct {
+	// Bucket is the name of the Google Cloud Storage bucket where the object
+	// is located.
+	Bucket string
+
+	// Object is the name and or path of the object stored in the bucket. It
+	// should not start with a forward slash.
+	Object string
+}
+
 type Service interface {
+	ParseURL(gsURL string) (*URL, error)
+
+	ReadArtifact(ctx context.Context, url string) ([]byte, error)
+	WriteArtifact(ctx context.Context, url, content string) error
```

Comment:
Should we make `content` here `[]byte` for consistency with the read method (which is also more generic than string)?

Reply:
Good idea. Updated